src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/IncomingCourtCaseComparator.java

```diff
@@ -63,48 +59,30 @@ private static boolean hasHearingDaysChanged(List<HearingDay> hearingDays, List<
     }
 
     private static boolean hasDefendantOffencesChanged(List<Defendant> defendants, List<Defendant> defendantsToCompare) {
-        for (Defendant defendantReceived : Collections.unmodifiableList(defendants)) {
-            for (Defendant existingDefendant : Collections.unmodifiableList(defendantsToCompare)) {
-                if (defendantReceived.getDefendantId().equals(existingDefendant.getDefendantId())) {
-                    if (areNotEqualIgnoringOrder(defendantReceived.getOffences(), existingDefendant.getOffences(), offenceComparator)) {
-                        return true;
-                    }
-                }
-            }
-        }
-        return false;
+        return Collections.unmodifiableList(defendants).stream()
```

Comment:
What's the reason for the conversion to unmodifiableList here?

Reply:
Just defensive coding that's all! not necessary at all.